src/procedure/nodes/select.cpp

```diff
@@ -264,25 +260,31 @@ bool SelectProcedureNode::execute(const ProcedureContext &procedureContext)
             // Check distance from reference site (if defined)
             if (distanceRef)
             {
-                r = procedureContext.configuration()->box()->minimumDistance(site.origin(), distanceRef->origin());
-                if (!inclusiveDistanceRange_.contains(r))
+                if (!inclusiveDistanceRange_.contains(
+                        procedureContext.configuration()->box()->minimumDistance(site.origin(), distanceRef->get().origin())))
                     continue;
             }
 
-            // All OK, so add site
-            sites_.push_back(&site);
+            // All OK, so add site and index in its stack (1...N numbering)
```

Comment:
Maybe not this exact wording, but...
```suggestion
            // All OK, so add site, its global index among all site stacks, and index in its stack (1...N numbering)
```